backend/corpora/dataset_processing/h5ad_data_file.py

```diff
@@ -29,21 +30,14 @@ class H5ADDataFile:
     def __init__(
         self,
         input_filename,
-        backed=False,
-        dataset_title=None,
-        dataset_about=None,
-        obs_index_column_name=None,
-        vars_index_column_name=None,
-        use_corpora_schema=True,
+        var_index_column_name=None,
     ):
         self.input_filename = input_filename
-        self.backed = backed
-        self.dataset_title = dataset_title
-        self.dataset_about = dataset_about
-        self.obs_index_column_name = obs_index_column_name
-        self.vars_index_column_name = vars_index_column_name
-
-        self.use_corpora_schema = use_corpora_schema
```

Comment:
Looking ahead a bit; the future of this code is that it will be moved out of the Portal repo and into the Explorer repo. It is worth noting that we may want to host datasets that do not adhere to the schema, but we want to manually transform them to CXG and put them up on the `/d/` dataroot. Because of this, I would recommend not removing this flag.

Reply:
@maniarathi Why would this be moved to the Explorer repo?  single-cell-explorer (hosted)?  I was thinking we might create a small, standalone CXG library, that encapsulates both the read and writing of the CXG format, and would also managed the CXG spec document.  Keeping all the logic in one place, and dealing with versioning compatibility seamlessly. Transforming from H5AD files to CXG could be part of that library as well.

Reply:
@atolopko-czi Yes, you're right. I had paused on reviewing this, but `title` and `about` are ones that we'd use as well. 

I'd do a quick check with Product to see if we have any potential datasets coming through the near term pipeline (say in the next month or so). If there is, I would recommend keeping it as is and opening a new issue to modify this to clean up the design during the Q4 cool down period. If not, then I would go ahead and get rid of it and let Product know that it'll take a touch more of time to convert non-compliant datasets the next time we want to. Does that sound like a reasonable compromise?

Reply:
Also @atolopko-czi -- creating a small standalone CXG library works too! A while back, Madison, Bruce, and I had discussed keeping it in the repo in which it serves (i.e. CXG serves the single-cell-explorer tool therefore should live there). I can't recall the exact details but worth discussing with them as well.

Reply:
If we need to manually create a CXG, we'd use the `single-cell-explorer` repo's [CLI](server/cli/convert_to_cxg.py), right? And since that repo has its own `h5ad_data_file.py` file, it will still support all the non-schema compliant H5AD->CXG conversion attributes (title, about, etc.).  So I think we should be able to make this repo only work for schema-compliant H5AD->CXG file.

Reply:
Ah shoot -- I had forgotten that the CLI in the `single-cell-explorer` still existed. This is actually duplicated code. The code here was pulled straight from there and then we just removed the CLI wrapper. 

I actually think that in the long term, we shouldn't have two copies of the code. The CLI in the `single-cell-explorer` is currently not used by anything. I see two options here: either consolidate the conversion code to a standalone repo or consolidate to the `single-cell-explorer` repo. The code should be able to handle both compliant and non-compliant H5ADs since (as you've stated below) any metadata that is schema-specific should probably not be in the CXG itself but rather surfaced perhaps through a microservice/API (imo).

Does that align with what you're think/does that make sense?

Reply:
Yup, 100%! The CLI, if needed, could tag along in the library repo (or subdir). And agree decisions need to be made re: attributes that are surfaced in CXG vs API.  I need to clean up my Arch notes so I can share & use as forum for ongoing discussion. :)